func-complete-check/check-marker/check-marker.v1.ts

```diff
@@ -0,0 +1,166 @@
+import * as RA from 'ramda-adjunct'
+import * as R from 'ramda'
+import { IAsyncTableService, AsyncTableService } from '../lib/storage-helper'
+import { ValidatedCheck } from '../typings/message-schemas'
+import moment from 'moment'
+import { ICheckFormService, CheckFormService } from '../lib/check-form.service'
+
+export interface ICheckMarkerFunctionBindings {
+  receivedCheckTable: Array<any>
+  checkNotificationQueue: Array<any>
+}
+
+declare interface Answer {
+  factor1: number
+  factor2: number
+  answer: string
+  sequenceNumber: number
+  question: string
+  clientTimestamp: string
+}
+
+declare interface CheckFormQuestion {
+  f1: number
+  f2: number
+}
+
+declare interface MarkingData {
+  answers: Array<Answer>
+  formQuestions: Array<CheckFormQuestion>
+  results: any
+}
+
+declare interface Mark {
+  mark: number
+  maxMarks: number
+  processedAt: Date
+}
+
+export class CheckMarkerV1 {
+
+  private _tableService: IAsyncTableService
+  private _sqlService: ICheckFormService
+
+  constructor (tableService?: IAsyncTableService, sqlService?: ICheckFormService) {
+    if (tableService === undefined) {
+      this._tableService = new AsyncTableService()
+    } else {
+      this._tableService = tableService
+    }
+
+    if (sqlService === undefined) {
+      this._sqlService = new CheckFormService()
+    } else {
+      this._sqlService = sqlService
+    }
+  }
+
+  async mark (functionBindings: ICheckMarkerFunctionBindings): Promise<void> {
+
+    const validatedCheck = this.findValidatedCheck(functionBindings.receivedCheckTable)
+    const markingData = await this.validateData(functionBindings, validatedCheck)
+    if (markingData === undefined) {
+      functionBindings.checkNotificationQueue.push({
+        checkCode: 'checkCode',
```

Comment:
I may be missing something here - but this message doesn't seem to have the actual checkCode in the message.

Reply:
yep, that's definitely a typo.  there is nothing consuming those messages, as we have not implemented the check notification service yet.  will fix